### ISSUE TITLE
Region.paddingHorizontal and Region.paddingVertical do not have the expected effect based on Region.paddingAll

### ISSUE THREAD

**Issue by nlbuescher** [edvin/tornadofx#936]
when I set `paddingAll` on something like a `vbox`:

```kotlin
vbox {
    paddingAll = 10
}
```

I get a `VBox` with a padding of 10 on each side. But when I set `paddingHorizontal`:

```kotlin
vbox {
    paddingHorizontal = 10
}
```

I would expect to get a VBox with a padding of 10 each on the left and the right, but instead I get a padding of *5* on each side.

**Comment by edvin** [comment]
At the time of creation we elected to let the value represent the total horizontal padding, evenly distributed on the two sides. In hindsight this might have been a bad decision. Does anybody else have an opinion about this? Should we change it?

**Comment by nlbuescher** [comment]
One issue that I can see is what to do with the getter. If I set `paddingLeft = 10` and `paddingRight = 15`, getting `paddingHorizontal` (returning 12.5) doesn't actually yield any useful information IMO. However, returning the total padding (`left + right`) on get and setting each side to the specified value on set would be confusing as well, eg:
```kotlin
var Region.paddingHorizontal
    get() = padding.left + padding.right
    set(value) {
        padding = padding.copy(horizontal = value)
    }

class TestView : View() {
    override val root = vbox {
        paddingHorizontal = 10
    }
    init {
        println(paddingHorizontal) // prints "20.0" (inconsistent)
    }
}
```

`paddingAll`, `paddingHorizontal`, and `paddingVertical` seem like really good candidates for write-only properties, but those aren't supported in Kotlin, although there *is* a neat tick with `@Deprecated(level = DeprecationLevel.ERROR)` that one could use with the getter that would actually disallow use of the getter at compile time (rather than `throw`ing at run time). That would look like this:

```kotlin
var Region.paddingHorizontal: Number
    @Deprecated("Property is write-only", level = DeprecationLevel.ERROR)
    get() = throw NotImplementedError() // never called at runtime, but technically returns a "Nothing" so the type system doesn't yell at us
    set(value) {
        padding = padding.copy(horizontal = value)
    }
```

**Comment by edvin** [comment]
Yeah, it looks like whatever we choose, some will be surprised by the result, so it might be best to simply leave it as it is. At least now horizontal will mirror left/right, so there is some kind of symmetry to it.
